src/main/java/org/apache/maven/plugins/shade/DefaultShader.java

```diff
@@ -48,35 +49,49 @@
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipException;
 
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.collections4.MultiValuedMap;
+import org.apache.commons.collections4.multimap.HashSetValuedHashMap;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugins.shade.filter.Filter;
 import org.apache.maven.plugins.shade.relocation.Relocator;
 import org.apache.maven.plugins.shade.resource.ManifestResourceTransformer;
 import org.apache.maven.plugins.shade.resource.ReproducibleResourceTransformer;
 import org.apache.maven.plugins.shade.resource.ResourceTransformer;
-import org.codehaus.plexus.component.annotations.Component;
-import org.codehaus.plexus.logging.AbstractLogEnabled;
 import org.codehaus.plexus.util.IOUtil;
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassVisitor;
 import org.objectweb.asm.ClassWriter;
 import org.objectweb.asm.commons.ClassRemapper;
 import org.objectweb.asm.commons.Remapper;
-
-import com.google.common.collect.HashMultimap;
-import com.google.common.collect.Multimap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * @author Jason van Zyl
  */
-@Component( role = Shader.class, hint = "default" )
+@Singleton
+@Named
 public class DefaultShader
-    extends AbstractLogEnabled
     implements Shader
 {
     private static final int BUFFER_SIZE = 32 * 1024;
 
+    private final Logger logger;
+
+    public DefaultShader()
+    {
+        this( LoggerFactory.getLogger( DefaultShader.class ) );
+    }
+
+    public DefaultShader( final Logger logger )
+    {
+        this.logger = Objects.requireNonNull( logger );
```

Comment:
Why so complicated instead of `...logger = LoggerFactory...`?

Reply:
UTs are mocking logger and use it to verify output, (did not modify UT more than I had). In plugin (runtime) the default ctor is used.